src/metrology_apis/__init__.py

```diff
@@ -1,15 +1,47 @@
 """Metrology APIs."""
 
-from typing import Final, Protocol, Self, override, runtime_checkable
+from typing import Any, Final, Generic, Protocol, Self, TypeVar, override, runtime_checkable
 
 import optype as op
 
 __version__: Final = "0.0.1.dev0"
 __all__ = ["__version__", "Dimension", "Quantity", "Unit"]
 
 
+VT = TypeVar('VT')
+DT = TypeVar('DT', bound='Dimension')
+UT = TypeVar('UT', bound='Unit[DT]')
+
+@runtime_checkable
+class MetrologyNamespace[Q: Quantity[VT, UT, DT], V, U: Unit[DT], D: Dimension](Protocol):
+
+    @staticmethod
+    def asdimension(obj: str | D) -> D: ...
+
+    @staticmethod
+    def asunit(obj: str | U) -> U: ...
+
+    @staticmethod
+    def asquantity(obj: Q | V, *, unit: U) -> Q: ...
+
+
 @runtime_checkable
 class Dimension(Protocol):
+    def __metrology_namespace__[Q: Quantity[VT, UT, DT], V, U: Unit[DT]](
+        self, /, *, api_version: str | None = None
+    ) -> MetrologyNamespace[Q, V, U, Self]:
+        """
+        Returns an object that has all the metrology API functions on it.
+        Parameters
+        ----------
+        api_version: str or None
+            string representing the version of the metrology API specification to be returned. If it is `None`, it should return the namespace corresponding to latest version of the metrology API specification.  If the given version is invalid or not implemented for the given module, an error should be raised. Default: `None`.
+        Returns
+        -------
+        out: Any
+            an object representing the metrology API namespace. It should have every top-level function defined in the specification as an attribute. It may contain other public names as well, but it is recommended to only include those names that are part of the specification.
```

Comment:
```suggestion
            An object representing the metrology API namespace. It
            should have every top-level function defined in the
            specification as an attribute. It may contain other public
            names as well, but it is recommended to only include those
            names that are part of the specification.
```